src/pip/_internal/commands/cache.py

```diff
@@ -184,7 +184,24 @@ def remove_cache_items(self, options: Values, args: List[Any]) -> None:
         for filename in files:
             os.unlink(filename)
             logger.verbose("Removed %s", filename)
+
+        http_dirs = filesystem.subdirs_with_no_files(self._cache_dir(options, "http"))
+        wheel_dirs = filesystem.subdirs_with_no_files(
+            self._cache_dir(options, "wheels")
+        )
+        dirs = list(http_dirs) + list(wheel_dirs)
+        for dirname in dirs:
```

Comment:
Use `itertools.chain` here

Reply:
as in something like this?:

```python
dirs = itertools.chain(http_dirs, wheel_dirs)
for dirname in dirs:
    ...
```

(Note that `dirs` is used on line 204, so I can't get away with `for dirname in itertools.chain(...)` as easily.)

Reply:
Not exactly because you can’t use `len` on an iterator.

Something like this

```python
for dirname in itertools.chain(http_dirs, wheel_dirs):
    ...

...

logger.info("Empty directories removed: %s", len(http_dirs) + len(wheel_dirs))
```

Reply:
Done. 👍

Reply:
So, turns out `http_dirs` and `wheel_dirs` are defined with results from `filesystem.subdirs_with_no_files(path_str)` which returns a Generator, so `len(http_dirs)` and `len(wheel_dirs)` won't work:

```
src/pip/_internal/commands/cache.py:204: error: Argument 1 to "len" has
incompatible type "Generator[str, None, None]"; expected "Sized"  [arg-type]
    ...logger.info("Empty directories removed: %s", len(http_dirs) + len(whee...
                                                        ^
```

Serves me right for not having a local dev environment and just assuming it'd work because it looked right. 😂